identity/src/main/java/com/stripe/android/identity/navigation/ConfirmationFragment.kt

```diff
@@ -31,58 +32,45 @@ internal class ConfirmationFragment(
         identityViewModelFactory
     }
 
-    private lateinit var binding: ConfirmationFragmentBinding
-
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View {
-        binding = ConfirmationFragmentBinding.inflate(inflater, container, false)
-        binding.kontinue.setOnClickListener {
-            lifecycleScope.launch {
-                identityViewModel.analyticsState.collectLatest {
-                    identityViewModel.sendAnalyticsRequest(
-                        identityViewModel.identityAnalyticsRequestFactory.verificationSucceeded(
-                            isFromFallbackUrl = false,
-                            scanType = it.scanType,
-                            requireSelfie = it.requireSelfie,
-                            docFrontRetryTimes = it.docFrontRetryTimes,
-                            docBackRetryTimes = it.docBackRetryTimes,
-                            selfieRetryTimes = it.selfieRetryTimes,
-                            docFrontUploadType = it.docFrontUploadType,
-                            docBackUploadType = it.docBackUploadType,
-                            docFrontModelScore = it.docFrontModelScore,
-                            docBackModelScore = it.docBackModelScore,
-                            selfieModelScore = it.selfieModelScore
+    ) = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+            val verificationPage by identityViewModel.verificationPage.observeAsState()
+            ConfirmationScreen(verificationPage) {
+                lifecycleScope.launch {
+                    identityViewModel.analyticsState.collectLatest {
```

Comment:
I’m a bit confused by this. In `identityViewModel.sendSucceededAnalyticsRequestForNative()`, it looks like we already call this exact same code. Why do we need to `collectLatest` here then?

Reply:
nice catch... I must be forgetting removing the extra lines 😃